### DISCUSSION TITLE
Can you use Keystone to retrieve data from other sources?

### DISCUSSION THREAD

**Discussion by biapar** [keystonejs/keystone#7962] [Questions]
Is't KeyStone a good choice for a customer private dashboard that get data from other sources?

**Comment by flexdinesh** [comment]
Keystone's virtual field is built specifically for this purpose - to fetch data from external sources. Check it out here: https://keystonejs.com/docs/guides/virtual-fields.

**Comment by rankun203** [comment]
Data retrieved from virtual field won't be available on the dashboard, we are trying to make those fields available on the dashboard for view / edit (CRUD).

Related discussion https://github.com/keystonejs/keystone/discussions/8319
